Reservation objet

EdouardDelep opened the pull request:
ce que j'ai commencé à faire

LinkinSky committed:
debut du projet reservation d'objets

LinkinSky committed:
ajout d'objet / suppression d'objet / début ajout réservation  / liste réservations

LinkinSky committed:
mini correctif

GregoireGiraud commented on ajouter_objet.php:
```diff
@@ -0,0 +1,13 @@
+    } catch(Exeption $e) {
+    die('erreur:'.$e->getMessage());
+}
+			$requete='INSERT INTO item VALUES(NULL,"'.$_POST['object_name'].'","'.$_POST['object_description'].'","'.$_POST['object_quantity'].'","'.$_POST['object_owner'].'")';
```
Il faut utiliser des requetes préparées

GregoireGiraud commented on ajouter_objet.php:
```diff
@@ -0,0 +1,13 @@
+<?php
```
Il faut include le _header.php (authentification, etc ...)

GregoireGiraud commented on reservation.php:
```diff
@@ -0,0 +1,172 @@
+$items = $requete_items->fetchAll();
+?>
+
+<table class="table">
```
[SYNTAXE] Une indentation à 4 espaces est mieux qu'une indentation à 2 espaces

GregoireGiraud commented on reservation.php:
```diff
@@ -0,0 +1,172 @@
+      <td><?= $items['quantity']; ?></td>
+      <td>
+      
+<!-- Réserver un objet -->
```
[SYNTAXE] L'indentation ne matche plus avec ce qu'il y a au dessus

GregoireGiraud commented on reservation.php:
```diff
@@ -0,0 +1,172 @@
+</form>
+      </td> 
+    </tr>
+    <?php } ?>
```
[SYNTAXE] le foreach se finit ici, mais c'est difficile de le voir, à cause de la mauvaise indentation

GregoireGiraud commented on reservation_admin.php:
```diff
@@ -0,0 +1,55 @@
+<?php
```
Cette page est pas mal faite, c'est bien !

GregoireGiraud commented on reservation_admin.php:
```diff
@@ -0,0 +1,55 @@
+
+try{
+    $DB = new PDO('mysql:host='.$confSQL['sql_host'].';dbname='.$confSQL['sql_db'].';charset=utf8',$confSQL['sql_user'],$confSQL['sql_pass'],array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
+    } catch(Exeption $e) {
```
[SYNTAXE] Tu peux juste désindenter d'un tab ici

GregoireGiraud commented on reservation_admin.php:
```diff
@@ -0,0 +1,55 @@
+
+$requete_reservations = $DB->prepare("SELECT * FROM reservation");
+$requete_reservations->execute();
+$reservations = $requete_reservations->fetchAll();
```
[AMELIORATION POSSIBLE] Vu qu'il n'y a pas de paramètres, pas besoin de préparer la requête ! (Même si ça marche bien) 
On peut utiliser query ici ;) 

$requete_reservations = $DB->query("SELECT * FROM reservation");
$reservations = $requete_reservations->fetchAll();

GregoireGiraud commented on reservation_admin.php:
```diff
@@ -0,0 +1,55 @@
+  </thead>
+  <tbody>
+  	<?php
+    $i=1;
```
[MENAGE] Pas besoin de mettre d'itérateur, il n'est jamais appelé !

GregoireGiraud commented on supprimer_objet.php:
```diff
@@ -0,0 +1,13 @@
+    } catch(Exeption $e) {
+    die('erreur:'.$e->getMessage());
+}
+			$requete="DELETE FROM item WHERE item_id={$_POST['object_id']}";
```
[SECURITE] Mieux vaut utiliser les requetes préparées 

$db->prepare("DELETE FROM item WHERE item_id=:item_id");
$db->execute("item_id" => $_POST['object_id']);

GregoireGiraud commented on supprimer_objet.php:
```diff
@@ -0,0 +1,13 @@
+<?php
```
[SECURITE] require _header.php

N'importe quelle personne dans le monde entier peut accéder à la page et supprimer un objet. La vérification sur l'identité est dans le _header.phpp

De plus, étant donné que tu ne fais pas de requêtes préparées, n'importe quelle personne dans le monde entier peut supprimer un objet MAIS SURTOUT PEUT FAIRE CE QU'ELLE VEUT SUR LA BASE DE DONNEES

LinkinSky committed:
Ajout des réservations / quelques ajustements

GregoireGiraud commented on reservation.php:
```diff
@@ -26,105 +26,67 @@
-      <th scope="col">Objet</th>
-      <th scope="col">Quantité</th>
-      <th scope="col">Action</th>
+      <th scope="col"><center>#</center></th>
```
[SYNTAXE] La balise html center est dépréciée, par contre, tu peux ajouter la classe text-center de bootstrap qui fait la même chose. 
Tu peux la mettre sur tout un tableau, sur toute une ligne, sur une cellule, etc ...
<table class="table text-center"> et ton tableau devrait être centré

GregoireGiraud commented on reservation.php:
```diff
@@ -26,105 +26,67 @@
+                </select>
+              </div>
+              <label for="exampleFormControlSelect1">Date de début</label>
+              <input type="datetime-local" class="form-control" name="reservation_start_date">
```
[SYNTAXE] On peut (doit) indenter le input dans le label

GregoireGiraud commented on reservation.php:
```diff
@@ -26,105 +26,67 @@
-<button type="submit" class="btn btn-danger">Supprimer</button>
-<input type="hidden" name="object_id" value="<?php echo $items['item_id']; ?>">
+  <input type="hidden" name="object_id" value="<?php echo $items['item_id']; ?>">
+  <button type="submit" class="btn btn-danger">Supprimer</button>
```
[AMELIORATION] Ce que tu fais ici marche très bien, dans une optique d'amélioration, on peut envoyer passer la suppresion en get

On n'a plus besoin de faire de formulaire, et un simple lien + bouton suffit très bien 
Exemple : <a href="url_delete.php?id=maValeur"<button>Supprimer</button></a>

ça marche, parce qu'en get on passe les infos par l'url (mapage.php?parametre=valeur). On peut donc recréer la requete get dynamiquement, et ne pas faire de vrai formulaire

LinkinSky committed:
Ajout du formulaire de modification des statuts pour les réservations + ajout des modifications, non fonctionnelles pour le moment (erreur de requête)

GregoireGiraud commented on ajouter_objet.php:
```diff
@@ -0,0 +1,16 @@
+    die('erreur:'.$e->getMessage());
+}
+
+			$requete='INSERT INTO item VALUES(NULL,"'.$_POST['object_name'].'","'.$_POST['object_description'].'","'.$_POST['object_quantity'].'","'.$_SESSION['login'].'")';
```
[SECURITE] Il faut faire des requêtes préparées. La page n'est pas du tout sécurisée actuellement. C'est dangereux d'avoir ça qui traine

GregoireGiraud replied:
[SYNTAXE] Il faut indenter (désindenter)

GregoireGiraud commented on ajouter_objet.php:
```diff
@@ -0,0 +1,16 @@
+
+try{
+    $DB = new PDO('mysql:host=localhost;dbname=payicam_accueil;charset=utf8','root','',array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
+    } catch(Exeption $e) {
```
[SYNTAXE] On peut désindenter 1 fois

GregoireGiraud commented on ajouter_reservation.php:
```diff
@@ -0,0 +1,17 @@
+
+try{
+    $DB = new PDO('mysql:host=localhost;dbname=payicam_accueil;charset=utf8','root','',array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
+    } catch(Exeption $e) {
```
[SYNTAXE] Désindenter 1 fois

GregoireGiraud commented on ajouter_reservation.php:
```diff
@@ -0,0 +1,17 @@
+
+}
+
+			$requete='INSERT INTO reservation VALUES(NULL,"'.$_SESSION['login'].'","'.$_POST['reservation_quantity'].'","'.$_POST['reservation_start_date'].'","'.$_POST['reservation_end_date'].'","'.$_POST['reservation_statue'].'","'.$_POST['object_id'].'")';
```
[SECURITE] Requête préparée
[SYNTAXE] Indentation

GregoireGiraud commented on modifier_reservation.php:
```diff
@@ -0,0 +1,26 @@
+
+try{
+    $DB = new PDO('mysql:host=localhost;dbname=payicam_accueil;charset=utf8','root','',array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
+    } catch(Exeption $e) {
```
[SYNTAXE] Désindenter 1 fois

GregoireGiraud commented on modifier_reservation.php:
```diff
@@ -0,0 +1,26 @@
+    $DB = new PDO('mysql:host=localhost;dbname=payicam_accueil;charset=utf8','root','',array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
+    } catch(Exeption $e) {
+    die('erreur:'.$e->getMessage());
+
```
[SYNTAXE] Saut de ligne inutile

GregoireGiraud commented on modifier_reservation.php:
```diff
@@ -0,0 +1,26 @@
+$requete_reservations = $DB->prepare("SELECT * FROM reservation");
+$requete_reservations->execute();
+$reservations = $requete_reservations->fetchAll();
+
```
[SYNTAXE] Revoir toute l'indentation après cette partie

GregoireGiraud commented on modifier_reservation.php:
```diff
@@ -0,0 +1,26 @@
+$requete_reservations->execute();
+$reservations = $requete_reservations->fetchAll();
+
+	if ($_POST['reservation_statue'] == 'En attente') { $statue == w; }
```
[AMÉLIORATION POSSIBLE] Pour tes 4 conditions, on peut améliorer tes conditions. On en reparlera en face

GregoireGiraud commented on modifier_reservation.php:
```diff
@@ -0,0 +1,26 @@
+$reservations = $requete_reservations->fetchAll();
+
+	if ($_POST['reservation_statue'] == 'En attente') { $statue == w; }
+	if ($_POST['reservation_statue'] == 'Accepter la réservation') { $statue == v; }
```
[NOMMAGE] On dit un statut, et non un statue

GregoireGiraud commented on reservation.php:
```diff
@@ -0,0 +1,132 @@
+
+try{
+    $DB = new PDO('mysql:host='.$confSQL['sql_host'].';dbname='.$confSQL['sql_db'].';charset=utf8',$confSQL['sql_user'],$confSQL['sql_pass'],array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
+    } catch(Exeption $e) {
```
[SYNTAXE] Désindenter

GregoireGiraud commented on modifier_reservation.php:
```diff
@@ -0,0 +1,26 @@
+
+}
+
+$requete_reservations = $DB->prepare("SELECT * FROM reservation");
```
[AMELIORATION POSSIBLE] Query au lieu de prepare

GregoireGiraud commented on reservation.php:
```diff
@@ -0,0 +1,132 @@
+include 'includes/header.php'; // insertion du fichier header.php : entête, barre de navigation
+
+$requete_items = $DB->prepare("SELECT * FROM item");
+$requete_items->execute();
```
[AMELIORATION POSSIBLE] Query au lieu de prepare

GregoireGiraud commented on reservation_admin.php:
```diff
@@ -0,0 +1,94 @@
+      <td><center><?= $reservations['end_date']; ?></center></td>
+      <td><center>
+                  <button type="button" class="
+                  <?php if ($reservations['status'] == 'w') {echo "btn btn-primary";}
```
[AMELIORATION / MENAGE] Il y a mieux à faire pour ne pas spammer le template, avec un array associant statut / classe BS

GregoireGiraud commented on reservation_admin.php:
```diff
@@ -0,0 +1,94 @@
+                        if ($reservations['status'] == 'a') {echo "btn btn-danger";}
+                        if ($reservations['status'] == 'f') {echo "btn btn-secondary";}
+                        ?>" data-toggle="modal" data-target="#exampleModalCenter">
+                  <?php if ($reservations['status'] == 'w') {echo "En attente";}
```
[AMELIORATION / MENAGE] Il y a mieux à faire pour ne pas spammer le template, avec un array associant statut / texte

GregoireGiraud commented on reservation_admin.php:
```diff
@@ -0,0 +1,94 @@
+                  </div>
+        </center></td>
+    </tr>
+    <?php $i++; } ?>
```
[MENAGE] Indentation inutile

GregoireGiraud commented on supprimer_objet.php:
```diff
@@ -0,0 +1,13 @@
+<?php
+try{
+    $DB = new PDO('mysql:host=localhost;dbname=payicam_accueil;charset=utf8','root','',array(PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION ));
+    } catch(Exeption $e) {
```
[SYNTAXE] Désindenter

GregoireGiraud commented on supprimer_objet.php:
```diff
@@ -0,0 +1,13 @@
+    die('erreur:'.$e->getMessage());
+}
+			$requete="DELETE FROM item WHERE item_id={$_POST['object_id']}";
+			$resultat = $DB->query($requete);
```
[SYNTAXE] Désindenter

GregoireGiraud reviewed (changes requested):
Il est nécessaire de traiter tous les aspects sécurité, syntaxe et ménage

LinkinSky committed:
Indentation + ajout du propriétaire de l'objet visible dans le tableau
Changement de suppression (via une variable visible/invisible) + suppression en méthode GET

LinkinSky committed:
Ajout d'objet et de réservation

EdouardDelep committed:
update

EdouardDelep committed:
Améliorations jusqu'au 04/01/2020

(Notamment, on a iconic et setFlashAndRedirect !!!!!! )

EdouardDelep committed:
Bouton supprimer objet visible uniquement pour les propriétaires de l'objet + changement de statuts avec des icones (l'ancien moyen de changement est en commentaire)

EdouardDelep committed:
Ajout des icônes d'états, possibilité d'annulation, message explicite de l'action lors que la souris passe sur le bouton

GregoireGiraud committed:
Merde du travail d'Edouard, conflits résolus

Status de résas mieux gérés, "droits" faits à l'arrache
